loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+
+If a plugin does not need to specify a language adapter other than the default language adapter, the plugin can be represented simply as the value string instead.
+
+### The `jars` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+A list of paths to nested JAR files to load, relative to the root directory inside of the mods JAR.
+
+### The `language_adapters` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs, where each key is the namespace of a language adapter and the value is an implementation of the `LanguageAdapter` interface.
+
+### The `depends` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Defines mods that this mod will not function without.
+
+A collection of `key: value` pairs, where each key is in the form of either `mavenGroup:modId` or `modId` and each value is either a [dependency object]() or an array of dependency objects. If an array of dependency objects is provided, the dependency matches if it matches ANY of the dependency objects.
+
+#### Dependency Object
+A dependency object is made up of a version specifier and a reason for that dependency. It may be represented in any of the following ways:
+```json5
+// As a JSON object, with both available fields
+{
+    version: "*",
+    reason: "Implements the same behavior as this mod."
+}
+```
+
+```json5
+// As a JSON object, omitting the optional "reason" field
+{
+    version: "*"
+}
+```
+
+```json5
+// As a single JSON string, containing only the version specifier
+"*"
+```
+
+#### Version Specifier
+A version range specifier can make use of any of the following patterns:
+* `*` — Matches any version. Will fetch the latest version available if needed
+* `1.0.0` — Matches exactly version 1.0.0 and no other versions
+* `=1.0.0` — Matches exactly version 1.0.0 and no other versions
+* `>=1.0.0` — Matches any version greater than or equal to 1.0.0
+* `>1.0.0` — Matches any version greater than version 1.0.0
+* `<=1.0.0` — Matches any version less than or equal to version 1.0.0
+* `<1.0.0` — Matches any version less than version 1.0.0
+* `1.0.x` — Matches any version with major version 1 and minor version 0.
+* `~1.0.0` — Matches the most recent version greater than or equal to version 1.0.0 and less than 1.1.0
+* `^1.0.0` — Matches the most recent version greater than or equal to version 1.0.0 and less than 2.0.0
+
+### The `breaks` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Defines mods that this mod either breaks or is broken by.
+
+A collection of `key: value` pairs, where each key is in the form of either `mavenGroup:modId` or `modId` and each value is either a [dependency object]() or an array of dependency objects. If an array of dependency objects is provided, the dependency matches if it matches ANY of the dependency objects.
```

Comment:
It would be useful if this had an optional `requires` when using a dependency object to define a mod breakage. This could be used for the sodium/indium case.
```json5
breaks: {
  sodium: {
    version: "*",
    reason: "Incompatible rendering mechanics",
    requires: "indium" // Could be a dependency object with version and/or use mavenGroup:modId
  }
}
```

Reply:
`requires` in the dependency object could be a bit confusing - implying sodium only breaks when indium is present.

`unless` would probably make it more clear.

Recursion could be a concern here but that is a thing for the dependency resolver to manage.

@AlexIIL I assume this wouldn't be too hard to model in the new partial-boolean solver. IF we consider this.

Reply:
Yes, this would probably be fairly simple to add to mod resolving.

(Although this is *also* possible in the original solver - partial-boolean *just* allows us to inspect the cause of an error, it doesn't actually add any new functionality).